sprites/tagalong.asm

```diff
@@ -0,0 +1,1614 @@
+    
+    LDY $11 : CPY.b #$0A : BEQ .proceed_to_just_draw
+    
+    CPX.b #$09 : BNE .not_overworld
+      ; Is the magic mirror being used?
+      CPY.b #$23 : BEQ .proceed_to_just_draw
+    .not_overworld
+    
+    ; Are we in text mode?
+    CPX.b #$0E : BNE .not_textbox
+    
+    CPY.b #$01 : BEQ .proceed_to_just_draw
+    CPY.b #$02 : BNE .not_textbox ; Text message mode
+    
+    .proceed_to_just_draw
+    
+    BRL .just_draw
+    
+    .not_textbox
+    
+    ; Make an exception for movement speeds. It's not dashing speed, so
+    ; I'm not sure what speed type this is.
+    LDA $5E : CMP.b #$04 : BEQ .dont_reset_link_speed
+      LDA.b #$0C : STA $5E
+    .dont_reset_link_speed
+    
+    JSR Tagalong_HandleTrigger ; $04A59E IN ROM
+    
+    SEP #$30
+    
+    LDA $7EF3CC : BNE .have_old_man_indeed
+      RTS
+    .have_old_man_indeed
+    
+    CMP.b #$04 : BNE .not_normal_old_man
+      LDX $02CF
+      LDA $1A50, X : BEQ .proceed_to_check_velocity 
+                     BMI .proceed_to_check_velocity
+      
+      LDA $02CF : CMP $02D3 : BEQ .proceed_to_check_velocity
+        BRL .advance_movement_index
+      .proceed_to_check_velocity
+      
+      BRL .check_link_velocity
+    
+    .not_normal_old_man
+    
+    LDA $4D : AND.b #$01 : BEQ .check_for_recoiling
+    
+    ; Is Link in a recoil state?
+    LDA $5D : CMP.b #$06 : BNE .check_for_recoiling ; if not, then branch away
+      LDA $02D3 : CMP $02CF : BNE .replace_that_tagalong
+      DEX : STX $02CF : BPL .replace_that_tagalong
+        LDA.b #$13 : STA $02CF
+    
+    .check_for_recoiling
+    
+    LDA $4D : AND.b #$02 : BEQ .check_link_velocity
+    
+    .replace_that_tagalong
+      
+      LDA $7EF3CC : TAX
+      
+      LDA Tagalong_OldMountainMan_ReplacementIds, X : STA $7EF3CC
+      
+      LDA.b #$40 : STA $02D2
+      
+      LDX $02CF
+      
+      LDA $1A00, X : STA $7EF3CD
+      LDA $1A14, X : STA $7EF3CE
+      
+      LDA $1A28, X : STA $7EF3CF
+      LDA $1A3C, X : STA $7EF3D0
+      
+      LDA $EE : STA $7EF3D2
+      
+      BRA Tagalong_UnusedOldMan
+    
+    .check_link_velocity
+    
+    LDA $30 : ORA $31 : BNE .link_moving
+      LDA $1A : AND.b #$03 : BNE .just_draw
+      LDA $02D3 : CMP $02CF : BEQ .just_draw
+      
+      SEC : SBC.b #$09 : BPL .no_back_wrap_a
+        CLC : ADC.b #$14
+      .no_back_wrap_a
+      
+      CMP $02CF : BNE .advance_movement_index
+      
+      BRL .just_draw
+    
+    .link_moving
+    
+    LDA $02D3 : SEC : SBC.b #$14 : BPL .no_back_wrap_b
+      CLC : ADC.b #$14
+    .no_back_wrap_b
+    
+    CMP $02CF : BNE .just_draw
+    
+    .advance_movement_index
+    
+    LDX $02CF : INX : CPX.b #$14 : BCC .no_wrap
+      LDX.b #$00
+    .no_wrap
+    
+    STX $02CF
+    
+    .just_draw
+    
+    BRL Tagalong_Draw
+    
+    RTS ; Unreachable
+}
+
+; ==============================================================================
+
+; \scawful: Potential expansion point for the Tagalong system.
```

Comment:
i like this idea of marking potential expansion places. would it make sense to mark it as \hacker or something?

Reply:
done